### PR TITLE
[Compiler-v2] fix timeout in third_party prover test

### PR DESCRIPTION
**Pull request by rahxephon89**
## Description

This PR resolves the timeout issue in the third_party prover test by exchanging between source_dep and dep when creating the move model. 

Also close [#13251](https://github.com/aptos-labs/aptos-core/issues/13251), close [#13252](https://github.com/aptos-labs/aptos-core/issues/13252)

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Removed `pragma verify=false` from the spec of `round` in `third_party/move/move-stdlib/sources/fixed_point32.move` and the prover test still passes.

## Key Areas to Review

When calling the prover in third_party and aptos, we need to make sure the same arguments passed to the move model builder, especially for `source_deps` and `deps`. 

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/move-prover/src/lib.rs [hunk 1]
```diff
@@ -73,7 +73,7 @@ pub fn run_move_prover_v2<W: WriteColor>(
     let now = Instant::now();
     let cloned_options = options.clone();
     let compiler_options = move_compiler_v2::Options {
-        dependencies: vec![],
```
Comment (rahxephon89) [inline third_party/move/move-prover/src/lib.rs:76]: See: https://github.com/aptos-labs/aptos-core/blob/5422dc17a873fa13faa15762cb7a7e22a5372ad5/third_party/move/tools/move-package/src/compilation/model_builder.rs#L155

**Review by brmataptos — APPROVED** [review]
LGTM.

File: third_party/move/move-prover/src/lib.rs [hunk 2]
```diff
@@ -83,7 +83,7 @@ pub fn run_move_prover_v2<W: WriteColor>(
         experiments: vec![],
         experiment_cache: Default::default(),
         sources: cloned_options.move_sources,
-        sources_deps: cloned_options.move_deps,
+        sources_deps: vec![],
```
Comment (wrwg) [inline third_party/move/move-prover/src/lib.rs:86]: What is this source_deps? Should have never been added to the prover, lets remove this later.

**Review by wrwg — APPROVED** [review]

File: third_party/move/move-prover/src/lib.rs (hunk 2 above)
Comment (rahxephon89) [reply third_party/move/move-prover/src/lib.rs:86]: This field is from`move_compiler_v2::Options`.

**Pull request merged**